rfcs/react-components/convergence/slot-children-render-function.md

```diff
@@ -0,0 +1,650 @@
+# RFC: Slot children render function
+
+---
+
+_@bsunderhus @ling1726 @layershifter_
+
+## TL;DR
+
+Slot children render function is a complex API that is not properly supported. To support this the best alternative so far,
+is to create a custom JSX pragma to ensure slot declaration will not lose any property
+and will be capable of properly rendering with children render function.
+
+## Summary
+
+By our [documentation](https://react.fluentui.dev/?path=/docs/concepts-developer-customizing-components-with-slots--page#replacing-the-entire-slot):
+
+> When you pass content or props to a slot, the component renders the content within a component or element based on the slot type.
+> If you need to replace the slot's entire content, including the containing element, pass a render function as the children.
+> This is an escape hatch in the slots API, so prefer the other techniques whenever possible.
+> If you replace the entire slot, accessibility, layout, and styling still work properly.
+> By passing `renderBigLetterIcon` as the `children`, the `span` that normally contains the icon is replaced with an `b` (bold).
+
+```tsx
+const renderBigLetterIcon (Component, props) => {
+  return <b>B</b>;
+};
+
+<Button icon={{ children: renderBigLetterIcon }}>Bold</Button>;
+```
+
+## Background
+
+There are 3 instances of properties that will be provided to a slot in different phases of the slot creation:
+
+1. External properties (`props`)
+2. Internal properties (`defaultProps`)
+3. Overrides
+
+### External properties
+
+External properties (`props`) are the properties provided by the client using a component with a specific slot.
+On the `AccordionHeader` example, the `button` slot external properties may be found as `props.button` inside the component's
+implementation.
+
+`props.button` is called a shorthand and can be either:
+
+1. External properties of the slot (an object)
+2. `string`, `number`, `Array` or `JSX element` value that will be provided to slot's `children`
+3. `null` or `undefined`
+
+```ts
+export const AccordionHeader: ForwardRefComponent<AccordionHeaderProps> = React.forwardRef((props, ref) => {
+  props.button; // this is the button shorthand
+});
+```
+
+### Internal properties
+
+Internal properties (`defaultProps`) are the properties provided by the `useState` hook, that should be overridden by
+External properties, those are the basic properties that ensure proper functioning of a slot but allowing the user
+to opt out of them.
+
+They can be declared when resolving the provided shorthand of a slot (`props.button`, in this example):
+
+```ts
+// useAccordionHeader
+const buttonShorthand = useARIAButtonShorthand(props.button, {
+  required: true,
+  // internal properties
+  defaultProps: {
+    disabled,
+    disabledFocusable,
```

Comment:
```suggestion
```

Let's remove them from the example as it's unclear from there `disabled` & `disabledFocusable` are coming